src/System.Private.ServiceModel/src/Internals/System/Runtime/Diagnostics/EtwDiagnosticTrace.cs

```diff
@@ -173,7 +173,47 @@ public TracePayload GetSerializedPayload(object source, TraceRecord traceRecord,
 
         public TracePayload GetSerializedPayload(object source, TraceRecord traceRecord, Exception exception, bool getServiceReference)
         {
-            throw ExceptionHelper.PlatformNotSupported();
+            string eventSource = null;
+            string extendedData = null;
+            string serializedException = null;
+
+            if (source != null)
+            {
+                eventSource = CreateSourceString(source);
+            }
+
+            if (traceRecord != null)
+            {
+                StringBuilder sb = StringBuilderPool.Take();
+                try
+                {
+                    using (StringWriter stringWriter = new StringWriter(sb, CultureInfo.CurrentCulture))
+                    {
+                        using (XmlWriter writer = XmlWriter.Create(stringWriter))
+                        {
+                            writer.WriteStartElement(DiagnosticStrings.ExtendedDataTag);
+                            traceRecord.WriteTo(writer);
+                            writer.WriteEndElement();
+                            writer.Flush();
+                            stringWriter.Flush();
+
+                            extendedData = sb.ToString();
+                        }
+                    }
+                }
+                finally
+                {
+                    StringBuilderPool.Return(sb);
```

Comment:
Do we need to StringBuilder.Clear() the StringBuilder before returning to the pool?